### PR TITLE
Impelement at_least, at_most, times, with, verify reset methods

### PR DESCRIPTION
**Pull request by ka2u**
Hi,

I try to implement at_least, at_most, times, with, verify, reset. I would like to use these methods. I based on Mockito, but if I can't along with your concept, you do revert or change.

Thank you.
Kazuhiro Shibuya

### CODE REVIEW

**Comment by masaki** [comment]
Thanks for your patch.
It's good that the following items are present.
- tests for at_most and times
- tests that "verify" will be FAIL
  - use Test::Exception ?

Thanks,

**Comment by ka2u** [comment]
Thank you. I will add some tests.

File: lib/Test/Double/Mock/Expectation.pm [hunk 1]
```diff
@@ -29,16 +49,70 @@ sub returns {
...
 
+sub _check_with {
+    my ($self, $counter, @args) = @_;
+    if (scalar @args != scalar @{$self->{with}}) {
+        $self->{check_with}->{$counter} = 0;
+        return;
+    }
+    for my $with_arg (@{$self->{with}}) {
+        for my $arg (@args) {
+            my ($ok, $stack) = cmp_details($arg, $with_arg);
```
Comment (masaki) [inline lib/Test/Double/Mock/Expectation.pm:69]: @ka2u 
I think, this logic cannot compare all values in `@{$self->{with}}` and `@args`.
looks like compare with[0] and args[0..n], isn't it?

**Comment by ka2u** [comment]
Yes, this is exactly right. Sorry, I close this pull request and I will send a request again.

**Pull request closed without merging**